includes/admin/class-sensei-extensions.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * File containing Sensei_Extensions class.
+ *
+ * @package Sensei\Admin
+ * @since 2.0.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei_Extensions class.
+ *
+ * All functionality pertaining to the admin area's extension directory.
+ *
+ * @since 2.0.0
+ */
+final class Sensei_Extensions {
+	const SENSEILMS_PRODUCTS_API_BASE_URL = 'https://senseilms.com/wp-json/senseilms-products/1.0';
+
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Courses constructor. Prevents other instances from being created outside of `Sensei_Extensions::instance()`.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Initializes the class and adds all filters and actions related to the extension directory.
+	 *
+	 * @since 2.0.0
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_assets' ) );
+		add_action( 'admin_menu', array( $this, 'add_admin_menu_item' ), 60 );
+	}
+
+	/**
+	 * Enqueues admin scripts when needed on different screens.
+	 *
+	 * @since  2.0.0
+	 * @access private
+	 */
+	public function enqueue_admin_assets() {
+		$screen = get_current_screen();
+		if ( in_array( $screen->id, array( 'sensei_page_sensei-extensions' ), true ) ) {
+			wp_enqueue_style( 'sensei-admin-extensions', Sensei()->plugin_url . 'assets/css/extensions.css', '', Sensei()->version, 'screen' );
+		}
+	}
+
+	/**
+	 * Call API to get Sensei extensions.
+	 *
+	 * @since  2.0.0
+	 *
+	 * @param  string $type      Product type ('plugin' or 'theme').
+	 * @param  string $category  Category to fetch (null = all).
+	 * @return array
+	 */
+	private function get_extensions( $type = null, $category = null ) {
+		$raw_extensions = wp_remote_get(
```

Comment:
Why not use `wp_safe_remote_get` here?

Reply:
6429912146ce51c10395f3384bca62c245dc6897